src/utils/uppy-config.ts

```diff
@@ -0,0 +1,47 @@
+import Uppy, { UppyFile } from '@uppy/core';
+import AwsS3Multipart, { AwsS3Part } from '@uppy/aws-s3-multipart';
+import {
+  abortMultipartUpload,
+  completeMultipartUpload,
+  createMultipartUpload,
+  createResourceAndMultipartUpload,
+  createResourceAndMultipartUploadResponseType,
+  listParts,
+  prepareUploadPart,
+} from '../api/fileApi';
+import { Uppy as UppyType } from '../types/file.types';
+
+interface UppyArgs {
+  uploadId: string;
+  key: string;
+}
+
+interface UppyPrepareArgs extends UppyArgs {
+  body: Blob;
+  number: number;
+}
+
+interface UppyCompleteArgs extends UppyArgs {
+  parts: AwsS3Part[];
+}
+
+export const createUppy = (resourceIdentifier: string, shouldAllowMultipleFiles: boolean): UppyType =>
+  Uppy<Uppy.StrictTypes>({
+    autoProceed: true,
+    restrictions: { maxNumberOfFiles: shouldAllowMultipleFiles ? null : 1 },
+  }).use(AwsS3Multipart, {
+    abortMultipartUpload: async (_: UppyFile, { uploadId, key }: UppyArgs) => await abortMultipartUpload(uploadId, key),
+    completeMultipartUpload: async (_: UppyFile, { uploadId, key, parts }: UppyCompleteArgs) =>
+      await completeMultipartUpload(uploadId, key, parts),
+    createMultipartUpload: async (file: UppyFile) => {
+      if (resourceIdentifier) return await createMultipartUpload(file);
+      else {
+        const response: createResourceAndMultipartUploadResponseType = await createResourceAndMultipartUpload(file);
```

Comment:
litt usikker på om denne hører til i `createUppy`...
Virker kanskje mer logisk at man sjekker om ressursen er opprettet før man prøver å starte opplasting, slik at man vet man har noe man vil laste opp når man har kommet hit. Hva skjer feks. om oppretting av ressurs feiler

Reply:
Vet ikke helt hvordan jeg skal få gjort det - da ressursen ikke skal opprettes før man starter opplasting. Man må ha et filnavn for å kunne opprette ressurs - og man må ha ressurs-id for å kunne starte opplasting :-(